File: joyal670-protenium-iro-hub-9d65cb9ae2c4/app/src/main/java/com/protenium/irohub/ui/main/dashboard/profile/adapter/SubscriptionAdapter.java
package com.protenium.irohub.ui.main.dashboard.profile.adapter;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.google.android.material.button.MaterialButton;
import com.protenium.irohub.R;
import com.protenium.irohub.model.dislike.getdislike.Tag;
import com.protenium.irohub.model.subscription_history.Subscription;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class SubscriptionAdapter extends RecyclerView.Adapter<SubscriptionAdapter.ViewHold> {

    private final List<Subscription> dataList;
    private final Context context;
    private SelectedId selectedId;

    public SubscriptionAdapter(Context fragment, List<Subscription> dataList) {
        this.dataList = dataList;
        this.context = fragment;
        if(fragment instanceof SelectedId){
            selectedId = (SelectedId) fragment;
        }
    }

    @NonNull
    @Override
    public ViewHold onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycerview_subscriptionhistory, parent, false);
        return new ViewHold(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHold holder, int position) {

        Log.e("TAG", "onBindViewHolder: " + dataList);
        Glide.with(context).load(dataList.get(position).getImage()).into(holder.ivSubscriptionHistory);

        holder.tvSubscriptionHistoryId.setText("#" + " " + dataList.get(position).getId());

        holder.tvStatus.setText(dataList.get(position).getPlanStatus());

        holder.tvSubscriptionHistoryCategoryPlan.setText(dataList.get(position).getMealPlanName() + " - " + dataList.get(position).getMealCategoryName());

        holder.rvsubscriptionHistoryViewDetailsBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                selectedId.setValue(dataList.get(position).getId());
            }
        });
    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }

    public class ViewHold extends RecyclerView.ViewHolder {

        @BindView(R.id.ivSubscriptionHistory)
        ImageView ivSubscriptionHistory;

        @BindView(R.id.tvSubscriptionHistoryId)
        TextView tvSubscriptionHistoryId;

        @BindView(R.id.tvStatus)
        TextView tvStatus;

        @BindView(R.id.tvSubscriptionHistoryCategoryPlan)
        TextView tvSubscriptionHistoryCategoryPlan;

        @BindView(R.id.rvsubscriptionHistoryViewDetailsBtn)
        MaterialButton rvsubscriptionHistoryViewDetailsBtn;


        public ViewHold(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            Log.e("TAG", "ViewHold: " + dataList);
        }
    }

    public interface SelectedId {
        void setValue(int selectedId);
    }

}

File: proteinium/app/src/main/java/com/proteinium/proteiniumdietapp/ui/main/home/myaccount/subscriptionhistory/adapter/SubscriptionAdapter.kt
package com.proteinium.proteiniumdietapp.ui.main.home.myaccount.subscriptionhistory.adapter

import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.RecyclerView
import coil.load
import com.proteinium.proteiniumdietapp.R
import com.proteinium.proteiniumdietapp.pojo.subscriptions_history.SubscriptionsHistory
import com.proteinium.proteiniumdietapp.preference.AppPreferences
import com.proteinium.proteiniumdietapp.ui.main.home.myaccount.subscriptionhistory.subscriptionhistorydetails.SubscriptionDetailedActivity
import com.proteinium.proteiniumdietapp.utils.CommonUtils
import kotlinx.android.synthetic.main.recycerview_subscriptionhistory.view.*

class SubscriptionAdapter(
    private var subscriptionHistoryList: List<SubscriptionsHistory>,
    private val isActive: Boolean,
    private val mContext: Context
) :
    RecyclerView.Adapter<SubscriptionAdapter.ViewHold>() {
    private var context: Context? = null

    class ViewHold(itemView: View) : RecyclerView.ViewHolder(itemView)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHold {
        context = parent.context
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.recycerview_subscriptionhistory, parent, false)
        return ViewHold(
            view
        )
    }

    override fun getItemCount(): Int {
        return subscriptionHistoryList.size
    }

    @RequiresApi(Build.VERSION_CODES.M)
    override fun onBindViewHolder(holder: ViewHold, position: Int) {
        holder.itemView.ivSubscriptionHistory.load(subscriptionHistoryList[position].image) {
            placeholder(R.drawable.ic_profile)
            error(R.drawable.ic_image_not)
        }
        if (AppPreferences.chooseLanguage == "ar") {
            holder.itemView.tvSubscriptionHistoryId.gravity = Gravity.END
        }
        subscriptionHistoryList[position].plan_status.also {
            holder.itemView.tvStatus.text = it
            if (AppPreferences.chooseLanguage == "ar" && subscriptionHistoryList[position].plan_status == "pending") {
                holder.itemView.tvStatus.text = context!!.getString(R.string.pending)
            }
        }


        when (subscriptionHistoryList[position].plan_status) {
//            context!!.getString(R.string.not_yet_started) -> {
//                holder.itemView.tvStatus.setTextColor(context!!.getColor(R.color.orange1))
//            }
//            context!!.getString(R.string.canceled) -> {
//                holder.itemView.tvStatus.setTextColor(context!!.getColor(R.color.green1))
//            }
//            context!!.getString(R.string.completed) -> {
//                holder.itemView.tvStatus.setTextColor(context!!.getColor(R.color.green))
//            }


        }
        holder.itemView.tvStartDate.text =
            context!!.getString(R.string.start_date) + " " + subscriptionHistoryList[position].plan_start_date
        holder.itemView.tvSubscriptionHistoryId.text =
            "#" + subscriptionHistoryList[position].id.toString()
        val categoryAndPlan =
            subscriptionHistoryList[position].meal_category_name + "-" + subscriptionHistoryList[position].meal_plan_name
        holder.itemView.tvSubscriptionHistoryCategoryPlan.text = categoryAndPlan
        holder.itemView.rvsubscriptionHistoryViewDetailsBtn.setOnClickListener {
            val intent = Intent(context, SubscriptionDetailedActivity::class.java)
            Log.e("meal_plan_sub_id", subscriptionHistoryList[position].id.toString())
            intent.putExtra("meal_plan_subscription_id", subscriptionHistoryList[position].id)
            intent.putExtra("is_renewal", isActive)
            context?.startActivity(intent)

        }
    }
}
